dist/angular-validation.js

```diff
@@ -346,6 +347,10 @@
                     var messageElem,
                         messageToShow = validMessage || $validationProvider.getDefaultMsg(validation).error;
 
+                    if(!ctrl.$error[ctrl.$name] && ctrl.$dirty && ctrl.$invalid){
```

Comment:
This check is necessary to guarantee that invalidFunc will be executed only by validators defined inside your validator´ directive (e.g. validator="required,email,minlength=16" )
It´s prevent weird behaviors in some cases when required message (from your validator) and ng-message (from angular async validator) will be show together, thus avoid unnecessary execution from the logic and allows your validation to integrate with any other native angular validation, e.g. you can use something like as follow:
<input type="text" name="inputname" ng-model="mymodel" required validator="email,minlength=16" ...